src/System.ComponentModel.TypeConverter/src/System/ComponentModel/TypeDescriptor.cs

```diff
@@ -1916,7 +1916,7 @@ private static ICollection PipelineFilter(int pipelineType, ICollection members,
                         }
                         catch (InvalidCastException)
                         {
-                            throw new ArgumentException(SR.GetResourceString(SR.TypeDescriptorExpectedElementType, typeof(Attribute).FullName));
+                            throw new ArgumentException(SR.Format(SR.TypeDescriptorExpectedElementType, typeof(Attribute).FullName));
```

Comment:
Apparently it's all to easy to make the mistake of using `GetResourceString(string resourceKey, string defaultString)` thinking you're getting formatting (like Console.Write for example). 

Does much code ever use the defaultString intentionally? Perhaps we should remove this parameter and offer `GetResourceStringWithDefault` so others don't make this mistake in future.

Reply:
@danmosemsft i could use same way to find all SR.GetResourceString and rename in GetResourceStringWithDefault. If you agree...is it better do that on other PR?or you can create a new issue and assign it to me if you want. Another observation is the use of string.Format instead of SR.Format, i haven't check all places, but maybe is better use SR.Format where possible.

Reply:
It is not clear to me what SR.Format gives over String.Format. Consistency is good, but I am not sure it is worth that cleanup.

Are many places using GetResourceString(string, string) correctly? 

Yes I agree would be a separate PR.

Reply:
i'll do some search.

Reply:
> It is not clear to me what SR.Format gives over String.Format

The main thing it gives is this:
https://github.com/dotnet/corefx/blob/master/src/Common/src/System/SR.cs#L43-L46
which I believe is currently only relevant on .NET Native / corert, where in certain builds resources may be stripped away entirely.

Reply:
Ah I wondered what that was for.